[AArch64] Generate BSP instead of TBL for select shuffles.

davemgreen opened the pull request:
In using BIF/BIT/BSL the constant mask has a larger chance of being regular, being able to be materialized with a movi. On some cpus the BIF/BIT/BSL is slightly quicker too.

sjoerdmeijer reviewed (approved):
LGTM

labrinea reviewed (approved):
The instruction sequences look better. Also BIF/BIT/BSL have higher throughput (4) than TBL (2), yet same execution latency (2). (At least on Neoverse V1 which I was looking at).

labrinea commented on llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:
```diff
@@ -13985,6 +13985,23 @@ SDValue AArch64TargetLowering::LowerVECTOR_SHUFFLE(SDValue Op,
+    SmallVector<SDValue> MaskElts;
+    for (int M : ShuffleMask)
+      MaskElts.push_back(
+          DAG.getConstant(M >= (int)NumElts ? 0 : 0xffffffff, dl, MVT::i32));
```
Nit: static cast here?

davemgreen committed:
[AArch64] Generate BSP instead of TBL for select shuffles.

In using BIF/BIT/BSL the constant mask has a larger chance of being regular,
being able to be materialized with a movi. On some cpus the BIF/BIT/BSL is
slightly quicker too.
